Branch-SDK/src/io/branch/referral/ServerRequestQueue.java

```diff
@@ -105,7 +104,10 @@ public void run() {
                     JSONObject json = jsonArr.getJSONObject(i);
                     ServerRequest req = ServerRequest.fromJSON(json, context);
                     if (req != null) {
-                        result.add(req);
+                        // No need to retrieve close or logout from previous session
+                        if (!(req instanceof ServerRequestRegisterClose || req instanceof ServerRequestLogout)) {
```

Comment:
Yes open will fire with the previous identity ID. Identity ID will rest only on a successful logout